tests/core/utils.py

```diff
@@ -1,11 +1,167 @@
+from smdebug.trials import create_trial
 
 
 def write_dummy_collection_file(trial):
     cm = CollectionManager()
     cm.create_collection("default")
     cm.add(Collection(trial))
     cm.export(trial, DEFAULT_COLLECTIONS_FILE_NAME)
+
+
+def delete_local_trials(local_trials):
+    for trial in local_trials:
+        shutil.rmtree(trial)
+
+
+# need this to seek to the right file offset for test output verification
+metrics_file_position = 0
+
+
+def check_metrics_file(save_steps, saved_scalars=None):
+    """
+    Check the SageMaker metrics file to ensure that all the scalars saved using
+    save_scalar(sm_metrics=True) or mentioned through SM_METRICS collections, have been saved.
+    """
+    global metrics_file_position
+    if is_sagemaker_job():
+        METRICS_DIR = os.environ.get(DEFAULT_SAGEMAKER_METRICS_PATH)
+        if not METRICS_DIR:
+            logging.warning("SageMaker Metric Directory not specified")
+            return
+        file_name = "{}/{}.json".format(METRICS_DIR, str(os.getpid()))
+        scalarnames = set()
+
+        import collections
+
+        train_metric = collections.defaultdict(list)
+        eval_metric = collections.defaultdict(list)
+
+        with open(file_name) as fp:
+            # since SM metrics expects all metrics to be written in 1 file, seeking to
+            # the right offset for the purpose of this test - so that the metrics logged in
+            # the corresponding test are verified
+            fp.seek(metrics_file_position)
+            for line in fp:
+                data = json.loads(line)
+                assert data["IterationNumber"] != -1  # iteration number should not be -1
+                metric_name = data["MetricName"]
+                if "TRAIN" in metric_name:
+                    train_metric[metric_name].append(data["IterationNumber"])
+                    scalarnames.add(metric_name.rstrip("_TRAIN"))
+                elif "EVAL" in metric_name:
+                    eval_metric[metric_name].append(data["IterationNumber"])
+                    scalarnames.add(metric_name.rstrip("_EVAL"))
+                else:
+                    scalarnames.add(
+                        metric_name.rstrip("_GLOBAL")
+                    )  # check the scalar saved using save_scalar()
+            metrics_file_position = fp.tell()
+        assert scalarnames
+
+        if saved_scalars:
+            assert len(set(saved_scalars) & set(scalarnames)) > 0
+
+        # check if all metrics have been written at the expected step number
+        for train_data in train_metric:
+            assert len(set(save_steps["TRAIN"]) & set(train_metric[train_data])) == len(
+                save_steps["TRAIN"]
+            )
+        for eval_data in eval_metric:
+            assert len(set(save_steps["EVAL"]) & set(eval_metric[eval_data])) == len(
+                save_steps["EVAL"]
+            )
+
+
+def check_trials(out_dir, save_steps, saved_scalars=None):
```

Comment:
can we improve the naming? check doesn't mean much, I would suggest "ensure written" or similar.

Reply:
If it's a private module function, prepend with underscore as per python convention,